### DISCUSSION TITLE
Cannot replicate Jekyll deployment of github pages locally

### DISCUSSION THREAD

**Discussion by WWWonderer** [skills/.github#331] [GitHub Pages] [answered]
Hello, 

I tried github pages here https://github.com/skills/github-pages, and from it generated my own repo here: https://github.com/WWWonderer/techblog. I want to be able to see my site/blog locally before pushing and committing to the repo and be auto-deployed by github pages. However, when trying locally, I am not getting the same result as the github deployment. 

github deployment:
![image](https://github.com/skills/.github/assets/34047699/54987c43-6715-4758-a43a-8d53a6e3f557)

local deployment results in a blank page. The posts under _posts folder are not shown and the theme is not applied. 

For local deployment testing, i have used 2 ways:

1. using the Docker container of github-pages as so:
`docker run -it --rm -v "<repo path on host os>:/usr/src/app" -p 4000:4000 starefossen/github-pages`

2. using ruby and jekyll by:
- installing `ruby`, `gem`, `bundle`, `jekyll`
- `jekyll build` and `jekyll serve`

Both these ways result in blank pages. I also tried modifying the `index.md` by adding 'hello' or something on it below the  front matter, and the 'hello' is rendered, so jekyll is indeed working, just not using the minima theme in `_config.yml` and not recognizing the posts. Also I verified that 'minima' is installed by gem by using `gem list`, and it indeed shows in the installed gems. 

There is no `_layouts`, `_includes` folders etc. in my repo, not even `Gemfile` etc. as in the [minima theme](https://github.com/jekyll/minima)  folder, so I kinda understand why the theme and posts may not be rendered on my local machine. However, this is the exact repo structure on github, how does github render the minima theme and the posts without a problem? What can I do to replicate my local environment such that it can replicate the behavior of github.io?

Thanks for any inputs!

**Comment by sinsukehlab** [comment, marked as answer]
GitHub Pages prepares you layouts and includes. You have to make a Gemfile etc. when trying local deployment.

**Reply by WWWonderer** [reply]
Thank you for the input! Yea in the end I made a new repo and started from scratch with ` jekyll new .`, this generates the Gemfile etc. and my local deployment was successful with it. I think maybe GitHub has a hidden Gemfile somewhere that is used but not uploaded by https://github.com/skills/github-pages, therefore making local deployment and replication impossible. If I'm right I suggest adding it. I'll delete my original techblog repo and close this discussion.

**Reply by sinsukehlab** [reply]
I'm glad that I could be of help.
